### PR TITLE
feat: allow disabling prisma migrations

### PR DESCRIPTION
**Pull request by Weakky**
Closes https://github.com/graphql-nexus/nexus/issues/371

This PR enable disabling migrations, which should be enough to support the introspection workflow.

### CODE REVIEW

File: src/lib/defaults.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    if (isObject(options[key])) {
+      result[key] = withDefaults(options[key], inputDefaults[key])
+    } else {
+      result[key] = options[key]
+    }
+  }
+
+  return result as U & T
+}
```
Comment (Weakky) [inline src/lib/defaults.ts:60]: I created a fancy helper to help us deal with all these options that has defaults. I think we should port that to Nexus and test it, I'm pretty happy with it, tell me what you think

File: src/worktime.ts [hunk 2]
```diff
@@ -17,8 +18,10 @@ if (process.env.LINK) {
  */
 export const PRISMA_QUERY_ENGINE_VERSION: string = require('@prisma/cli/package.json').prisma.version
 
-export const plugin: WorktimePlugin<Settings> = (_settings) => (p) => {
-  let elapsedMsSinceRestart = Date.now()
+export const plugin: WorktimePlugin<Settings> = (userSettings) => (p) => {
+  const settings = withDefaults(userSettings, {
+    migrations: true,
+  })
```
Comment (Weakky) [inline src/worktime.ts:24]: Here's the usage. The biggest win is the typings. It makes sure that when you pass a default, that the resulting type will put that default as non-optional.

File: src/worktime.ts [hunk 3]
```diff
@@ -213,21 +216,18 @@ export const plugin: WorktimePlugin<Settings> = (_settings) => (p) => {
...
     if (file.match(/.*schema\.prisma$/)) {
-      // Prevent from prompting twice when some updates to the schema are queued while the prompt is shown
-      const elapsed = Date.now() - elapsedMsSinceRestart
-      if (elapsed < 50) {
-        return
+      if (settings.migrations === true) {
+        await promptForMigration(p, watcher, file)
+      } else {
+        await runPrismaGenerators(p)
+        watcher.restart(file)
```
Comment (Weakky) [inline src/worktime.ts:226]: When migrations aren't enabled, we just run the prisma generators to regenerate the client. When using introspection, it means that whenever you introspect your database, you schema will be updated, which will regenerate the client and that should be all you need!

File: src/lib/defaults.ts (hunk 1 above)
Comment (jasonkuhrt) [reply src/lib/defaults.ts:60]: Is the goal to traverse just plain objects? Then maybe pull up a utility func for that (quite a few available). And then mention in the jsdoc that only objects are followed, not arrays, etc.

Comment (Weakky) [reply src/lib/defaults.ts:60]: The only utility func I know that has roughly the same behavior is https://lodash.com/docs/4.17.15#defaultsDeep which I didn't want to add to `nexus-plugin-prisma`. Do you know any other?

**Pull request merged**